### PR TITLE
The DRAGnet! A new tool for Security.

### PR DESCRIPTION
**Pull request by Cheridan**
Ordered by Skorvold!

http://i.imgur.com/ckAV0Qa.png

The DRAGnet has two firing modes. The primary firing mode fires an energy net in a wave in front of the user. If it hits a living target it does stamina damage, and places a bluespace field that will teleport them away 3 seconds. The teleport target is based on the target of the teleporter control console in the teleporter room. If the teleporter room isn't active, it'll teleport them a distance away instead.

The second firing mode is a temporary snare-trap. If someone walks over it, it'll snap on similarly to a bear trap. It vanishes when removed.

Also it's not fully finished yet, the sprites (made by Nienhaus) aren't plugged in yet. :)

### CODE REVIEW

**Comment by Jordie0608** [comment]
cheridumb

**Comment by Dorsisdwarf** [comment]
This seems pretty overpowered, and also pointless since there are no teleporter beacons to brig cells.

Also, the brig needs a holding cell area to hold the perps who get teleported in.

**Comment by Lumipharon** [comment]
Iif only beacons could be moved.

**Comment by Dorsisdwarf** [comment]
Oh and also the offitsers have to break into the teleporter to set this up/beg the AI/bribe the captain

**Comment by Fox-McCloud** [comment]
This is pretty much like Goon's teleporter gun, and I suspect that's what its value will end up being here too--that is to say traitors will value it way more than anyone else for its ability to teleport people to the middle of nowhere where they can never get out and just suffocate to death.

**Comment by Lumipharon** [comment]
It's an admin only weapon, getting teleporter access is not high on the list of issues with being able to use this.

**Comment by drynwyn** [comment]
Would be a great thing to add to R&D.

**Comment by Chocobro** [comment]
i'd really prefer if this was like mid/high on r&d stuff. otherwise :+1:

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 1]
```diff
@@ -12,6 +12,19 @@
 	select_fire(user)
 	update_icon()
 
+
+/obj/item/weapon/gun/energy/gun/dragnet
+	name = "DRAGnet"
+	desc = "The \"Dynamic Rapid-Apprehension of the Guilty\" net is a revolution in law enforcement technology."
+	icon_state = "dragnet"
+	item_state = null	//so the human update icon uses the icon_state instead.
+	origin_tech = "combat=4;magnets=4;materials=4"
```
Comment (GunHog) [inline code/modules/projectiles/guns/energy/nuclear.dm]: A teleportation device should have origins in Bluespace technology.

**Comment by Miauw62** [comment]
pls make it have its own teleporter toggle or something, it being based off the teleporter in the teleporter room is awful.

**Comment by Cheridan** [comment]
Working off beacons is the easiest and most consistent method. Asking the AI to set a teleporter target is not any harder than asking it to open doors.

**Comment by Fox-McCloud** [comment]
@Cheridan 

I generally agree, but don't you think it would be best if you could set it to work off of multiple base stations as well (much like the hand tele can)?

**Comment by Lumipharon** [comment]
Being able to lock it in is no fun, but yeah it would make sense to work like the hand tele, can choose which hub to take it's target from.

**Comment by Atskadan** [comment]
what if you made it lock onto a beacon by hitting the gun with the beacon

**Comment by GunHog** [comment]
Is this alive again!?

**Comment by WJohn** [comment]
This should honestly be a feature for abductors.

**Comment by Razharas** [comment]
If cheri wants to finish it he can reopen it

**Pull request closed without merging**